logfire/_internal/baggage.py

```diff
@@ -44,6 +48,17 @@ def set_baggage(**values: str) -> Iterator[None]:
     """
     current_context = context.get_current()
     for key, value in values.items():
+        if not isinstance(value, str):  # type: ignore
+            warnings.warn(
+                f'Baggage value for key "{key}" is a {type(value).__name__}. Converting to string.', stacklevel=3
+            )
+            value = logfire_json_dumps(value)
```

Comment:
We said in the call we are _not_ going to support non-strings for now

Reply:
I removed this, but now it sounds like you want it back?